packages/plugin-text/src/factory/editor.tsx

```diff
@@ -35,6 +38,19 @@ export const createTextEditor = (
         lastValue.current = props.state.value
       }
     }, [lastValue, props.state.value])
+
+    const pluginClosure = React.useRef({ overlayContext })
+    const [slatePlugins, setSlatePlugins] = React.useState<
```

Comment:
should be `React.useRef`. Otherwise, you'll cause one unecceary rerendering and also have a side effect in render

Reply:
Alternatively: you leave it in state but recreate the plugins if they change (i.e. by calling `setStatePlugins` in a `React.useEffect` with `options.plugins` as dep